gateway-js/src/__tests__/executeQueryPlan.test.ts

```diff
@@ -1636,4 +1636,640 @@ describe('executeQueryPlan', () => {
       });
     });
   });
+
+  describe('@requires', () => {
+    test('handles null in required field correctly (with nullable fields)', async () => {
+      const s1 = {
+        name: 'S1',
+        typeDefs: gql`
+          type T1 @key(fields: "id") {
+            id: Int!
+            f1: String
+          }
+        `
+      }
+
+      const s2 = {
+        name: 'S2',
+        typeDefs: gql`
+          type Query {
+            getT1s: [T1]
+          }
+
+          extend type T1 @key(fields: "id") {
+            id: Int! @external
+            f1: String @external
+            f2: T2 @requires(fields: "f1")
+          }
+
+          type T2 {
+            a: String
+          }
+        `
+      }
+
+      const { serviceMap, schema, queryPlanner} = getFederatedTestingSchema([ s1, s2 ]);
+
+      const s1_data = [
+        { id: 0, f1: "foo" },
+        { id: 1, f1: null },
+        { id: 2, f1: "bar" },
+      ];
+
+      addResolversToSchema(serviceMap['S1'].schema, {
+        T1: {
+          __resolveReference(ref) {
+            return s1_data[ref.id];
+          },
+        },
+      });
+
+      addResolversToSchema(serviceMap['S2'].schema, {
+        Query: {
+          getT1s() {
+            return [{id: 0}, {id: 1}, {id: 2}];
+          },
+        },
+        T1: {
+          __resolveReference(ref) {
+            // the ref has already the id and f1 is a require is triggered, and we resolve f2 below
+            return ref;
+          },
+          f2(o) {
+            return o.f1 === null ? null : { a: `t1:${o.f1}` };
+          }
+        }
+      });
```

Comment:
Nothing wrong here, but these resolvers can be added to the fixture object itself i.e.
`const s1 = { name, typeDefs, resolvers };`